### PR TITLE
unarc: init at 0-unstable-2020-06-05

### PR DESCRIPTION
**Pull request by lucasew**
## Description of changes
Introduces unarc with auto update and meta.mainProgram set up

Signed-off-by: lucasew <lucas59356@gmail.com>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lucasew** [comment]
Result of `nixpkgs-review pr 322687` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>unarc</li>
  </ul>
</details>

**Comment by lucasew** [comment]
Result of `nixpkgs-review pr 322687` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>unarc</li>
  </ul>
</details>

File: pkgs/by-name/un/unarc/package.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{ stdenv
+, unstableGitUpdater
```
Comment (FliegendeWurst) [inline pkgs/by-name/un/unarc/package.nix:2]: Please format with the new formatter.

File: pkgs/by-name/un/unarc/package.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    owner = "xredor";
+    repo = "unarc";
+    rev = "adc333d6cdd76d72da254cc80d766fbbcc683c95";
+    hash = "sha256-ysOei44P3K+aA+h73DuHlgwTKqQx/Xq8z+DefB6Qhcs=";
+  };
+
+  postPatch = ''
+    substituteInPlace 'CUI.h' \
+      --replace 'fgets(answer, 256, stdin);' 'if (!fgets (answer, 256, stdin)) return FALSE;' \
+      --replace 'printf (help)' 'printf ("%s", help)'
```
Comment (FliegendeWurst) [inline pkgs/by-name/un/unarc/package.nix:21]:
```suggestion
      --replace-fail 'fgets(answer, 256, stdin);' 'if (!fgets (answer, 256, stdin)) return FALSE;' \
      --replace-fail 'printf (help)' 'printf ("%s", help)'
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/un/unarc/package.nix (hunk 2 above)
Comment (lucasew) [reply pkgs/by-name/un/unarc/package.nix]: Done

File: pkgs/by-name/un/unarc/package.nix (hunk 1 above)
Comment (lucasew) [reply pkgs/by-name/un/unarc/package.nix]: Done

**Comment by lucasew** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 322687`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>melange</li>
    <li>terraform-providers.aiven</li>
    <li>terraform-providers.migadu</li>
    <li>terraform-providers.sentry</li>
    <li>unarc</li>
  </ul>
</details>

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**